src/system_identity/environment.rs

```diff
@@ -0,0 +1,82 @@
+use std::sync::LazyLock;
+
+use http::Uri;
+
+// Known endpoints. Should these be configurable instead? Otherwise,
+// if they change we will need to create a new release.
+// Staging endpoints
+static STAGING_TOKEN_RENEWAL_ENDPOINT: LazyLock<Uri> = LazyLock::new(|| {
+    const STAGING_TOKEN_RENEWAL_ENDPOINT_STR: &str =
+        "https://system-identity-oauth.staging-service.newrelic.com/oauth2/token";
+    Uri::try_from(STAGING_TOKEN_RENEWAL_ENDPOINT_STR)
+        .expect("Failed to parse known URL: STAGING_TOKEN_RENEWAL_ENDPOINT")
+});
+
+static STAGING_IDENTITY_CREATION_ENDPOINT: LazyLock<Uri> = LazyLock::new(|| {
+    const STAGING_IDENTITY_CREATION_ENDPOINT_STR: &str = "https://staging-api.newrelic.com/graphql";
+    Uri::try_from(STAGING_IDENTITY_CREATION_ENDPOINT_STR)
+        .expect("Failed to parse known URL: STAGING_IDENTITY_CREATION_ENDPOINT")
+});
+
+// EU endpoints
+static EU_TOKEN_RENEWAL_ENDPOINT: LazyLock<Uri> = LazyLock::new(|| {
+    const EU_TOKEN_RENEWAL_ENDPOINT_STR: &str =
+        "https://system-identity-oauth.service.newrelic.com/oauth2/token";
+    Uri::try_from(EU_TOKEN_RENEWAL_ENDPOINT_STR)
+        .expect("Failed to parse known URL: EU_TOKEN_RENEWAL_ENDPOINT")
+});
+static EU_IDENTITY_CREATION_ENDPOINT: LazyLock<Uri> = LazyLock::new(|| {
+    const EU_IDENTITY_CREATION_ENDPOINT_STR: &str = "https://api.eu.newrelic.com/graphql";
+    Uri::try_from(EU_IDENTITY_CREATION_ENDPOINT_STR)
+        .expect("Failed to parse known URL: EU_IDENTITY_CREATION_ENDPOINT")
+});
+
+// US endpoints
+static US_TOKEN_RENEWAL_ENDPOINT: LazyLock<Uri> = LazyLock::new(|| {
+    const US_TOKEN_RENEWAL_ENDPOINT_STR: &str =
+        "https://system-identity-oauth.service.newrelic.com/oauth2/token";
+    Uri::try_from(US_TOKEN_RENEWAL_ENDPOINT_STR)
+        .expect("Failed to parse known URL: US_TOKEN_RENEWAL_ENDPOINT")
+});
+static US_IDENTITY_CREATION_ENDPOINT: LazyLock<Uri> = LazyLock::new(|| {
+    const US_IDENTITY_CREATION_ENDPOINT_STR: &str = "https://api.newrelic.com/graphql";
+    Uri::try_from(US_IDENTITY_CREATION_ENDPOINT_STR)
+        .expect("Failed to parse known URL: US_IDENTITY_CREATION_ENDPOINT")
+});
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum SystemIdentityCreationEnvironment {
+    US,
+    Staging,
+    EU,
+    Custom {
+        token_renewal_endpoint: Uri,
+        system_identity_creation_uri: Uri,
+    },
+}
+
+impl SystemIdentityCreationEnvironment {
+    pub fn identity_creation_endpoint(&self) -> &Uri {
+        match self {
+            SystemIdentityCreationEnvironment::US => &US_IDENTITY_CREATION_ENDPOINT,
+            SystemIdentityCreationEnvironment::Staging => &STAGING_IDENTITY_CREATION_ENDPOINT,
+            SystemIdentityCreationEnvironment::EU => &EU_IDENTITY_CREATION_ENDPOINT,
+            SystemIdentityCreationEnvironment::Custom {
+                system_identity_creation_uri,
+                ..
+            } => system_identity_creation_uri,
```

Comment:
Why not hardcode the strings there and call the `try_from` at the end?
Same for the `token_renewal_endpoint`.

Reply:
Oh, I see your point. Let it stay then. No big deal. My reasoning is totally subjective. I would say this is premature optimization. In my head, we should most of the times use the hardcoded strings. The custom should not be used unless in testing or a change in the domain. I wouldn't expect also users to be calling all the time. Maybe I'm assuming too much. It also adds a bit of delay to the `LazyLock`.

Anyway, the code is well contained in that file. No big deal. We have no real usage metrics. So... Let's keep it as is! Thanks for your explanation!